src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1236,6 +1237,55 @@ private bool InCustomNode()
         }
 
         /// <summary>
+        /// Display a label for geometry based on the paths.
+        /// </summary>
+        public override void AddLabelForPath(string path)
+        {
+            // make var_guid from var_guid:0:1
+            var nodePath = path.Contains(':') ? path.Remove(path.IndexOf(':')) : path;
+            var labelName = nodePath + TextKey;
+            lock (Model3DDictionaryMutex)
+            {
+                // first, remove current labels of the node
+                // it does not crash if there is no such key in dictionary
+                Model3DDictionary.Remove(labelName);
+                
+                // second, add requested labels
+                var textGeometry = HelixRenderPackage.InitText3D();
+                var bbText = new BillboardTextModel3D
+                {
+                    Geometry = textGeometry
+                };
+
+                // it may be requested an array of items to put labels
+                // for example, the first item in 2-dim array - path will look like var_guid:0
+                // and it will select var_guid:0:0, var_guid:0:1, var_guid:0:2 and so on
+                var nodeLabelKeys = Model3DDictionary.Keys.Where(k => k.StartsWith(path + ':')).ToList();
+                foreach (var key in nodeLabelKeys)
```

Comment:
What for we create `textGeometry`  and `bbText`, if there won't be any nodeLabelKeys.

Imagine next situation:
![image](https://cloud.githubusercontent.com/assets/8158404/13458636/15a436b2-e077-11e5-8129-d6c3c1af597c.png)

We will call `AttachAllGeometryModel3DToRenderHost`, check all geometry, create `textGeometry`  and `bbText. But it's not needed, since there is no geometry to select (it's just number selected).

Can you add some check before, if there wasn't any geometry item clicked, then we won't process further.